### PR TITLE
Extending type coverage for vector

### PR DESCRIPTION
**Pull request by kopylovanat**
Added coverage for types byte, int8_t, int16_t, int32_t, int64_t,
                               uint8_t, uint16_t, uint32_t, uint64_t
Moved some duplicated code to common file

Not in this commit:
                type coverage for as(), convert(), get_count(), get_size()
		                  lo(), hi(), odd(), even()
                test coverage:
		  operator vector_t()
	          vec(const argTN&... args)
                  dataT()
                  convert() with different roundingMode
                  results check for convert()
                  results check for as()
                  different permutations for swizzle()
                  operator return type for swizzle() return type
                  swizzle(), XYZW_SWIZZLE, RGBA_SWIZZLE of __swizzled_vector__
                  swizzle(), XYZW_SWIZZLE, RGBA_SWIZZLE tests with non-zero value
                  XYZW_ACCESS, RGBA_ACCESS, INDEX_ACCESS
                  lo(), hi(), odd(), even() tests with non-zero value
                  load(), store() tests with non-zero value
                  load(), store() with different adressSpace
                  operator=
                  const-correctness check for operators
                  reliable control values for operators tests
                  possible better result checks for operators
                  check for return type for operators

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -78,6 +78,38 @@ function(get_opencl_type OUT_LIST)
...
+function(get_fixed_width_type OUT_LIST)
+  set(FIXED_WIDTH_LIST "")
+  list(APPEND FIXED_WIDTH_LIST std::int8_t
+    std::int16_t
+    std::int32_t
+    std::int64_t
+    std::uint8_t
+    std::uint16_t
+    std::uint32_t
+    std::uint64_t)
```
Comment (keryell) [inline tests/CMakeLists.txt:97]: First time I see some C++ code inside a CMake file. :-)

File: tests/common/common_python_vec.py [hunk 2]
```diff
@@ -192,6 +219,12 @@ class ReverseData:
 }
 """)
 
+def clear_type_str(type_str):
+    """
+    Clear type name from from namespaces ans whitspaces
```
Comment (keryell) [inline tests/common/common_python_vec.py:224]: Spell checker...

**Review by keryell — CHANGES_REQUESTED** [review]
Big contribution.
Just a minor point.

**Review by ProGTX — APPROVED** [review]

File: tests/common/common_python_vec.py (hunk 2 above)
Comment (keryell) [reply tests/common/common_python_vec.py:224]: Perhaps the function name could be more understandable.

File: tests/common/vector.template [hunk 3]
```diff
@@ -38,3 +39,4 @@ class TEST_NAME : public util::test_base {
 util::test_proxy<TEST_NAME> proxy;
 
 } /* namespace vector_$CATEGORY_$TYPE_NAME__ */
+$ENDIF
```
Comment (keryell) [inline tests/common/vector.template:42]: Missing end-of-line.

File: tests/common/vector_swizzles.template [hunk 4]
```diff
@@ -69,3 +69,4 @@ class TEST_NAME : public util::test_base {
 util::test_proxy<TEST_NAME> proxy;
 
 } /* namespace vector_swizzles_$TYPE_NAME__ */
+$ENDIF
```
Comment (keryell) [inline tests/common/vector_swizzles.template:72]: Missing end-of-line.

File: tests/common/common_python_vec.py (hunk 2 above)
Comment (kopylovanat) [reply tests/common/common_python_vec.py]: Fixed spelling. Renamed to  `remove_namespaces_whitspaces`

File: tests/common/vector.template (hunk 3 above)
Comment (kopylovanat) [reply tests/common/vector.template]: Fixed.

File: tests/common/vector_swizzles.template (hunk 4 above)
Comment (kopylovanat) [reply tests/common/vector_swizzles.template]: Fixed.

File: tests/common/common_python_vec.py [hunk 5]
```diff
@@ -219,9 +219,9 @@ class ReverseData:
 }
 """)
 
-def clear_type_str(type_str):
+def remove_namespaces_whitspaces(type_str):
     """
-    Clear type name from from namespaces ans whitspaces
+    Clear type name from namespaces and whitspaces
```
Comment (keryell) [inline tests/common/common_python_vec.py:224]: I guess you mean `whitespaces` or `white_spaces` everywhere in the code and comments.

File: tests/common/common_python_vec.py [hunk 6]
```diff
@@ -236,7 +236,7 @@ def wrap_with_kernel(type_str, kernel_name, test_name, test_string):
 
     return wrap_with_extension_checks(type_str,
                                       kernel_template.substitute(
-                                          kernelName=clear_type_str(kernel_name),
+                                          kernelName=remove_namespaces_whitspaces(kernel_name),
```
Comment (keryell) [inline tests/common/common_python_vec.py:239]: Curious indenting

**Review by keryell — CHANGES_REQUESTED** [review]
Almost done!

File: tests/common/common_python_vec.py (hunk 5 above)
Comment (kopylovanat) [reply tests/common/common_python_vec.py]: Yes, missed it, thank you! fixed.

File: tests/common/common_python_vec.py (hunk 6 above)
Comment (kopylovanat) [reply tests/common/common_python_vec.py]: Fixed.

**Review by keryell — APPROVED** [review]
Thanks!

**Pull request merged**